### PR TITLE
[dep] Upgrade axios from 0.21.4 to 1.6.0

### PR DESCRIPTION
**Pull request by Drarig29**
### What and why?

Closes https://github.com/DataDog/datadog-ci/issues/1096

Bumping `axios` to `1.6.0` came with 2 issues:
- https://github.com/axios/axios/discussions/5866, fixed in a5c801aba9fa8131aeefc4ffa75002df67156c8e
- Sending requests with `FormData` (which sends `multipart/form-data` payloads) was failing silently in our backend with only a warning.
  - For reference, the warning was `Decoder has partially decoded field`, which was probably caused by an incorrect boundary. Bumping `form-data` from `3.0.1` to `4.0.0` ([compare changes](https://github.com/form-data/form-data/compare/v3.0.1...v4.0.0)) solved the issue: 41ce71b54e19deacf6953865dd14ee3031cf61c4
  - This bug was caught in [one of our e2e tests](https://github.com/DataDog/datadog-ci/actions/runs/6704428841/job/18217012787#step:9:43), for the `junit upload` command. I ensured this command also works with the standalone binary. I also tested with `synthetics upload-application`, which is another command using the  `form-data` package.

### How?

Bump `axios` and `form-data`, and add `node_modules/axios/dist/node/axios.cjs` to the `pkg.assets` to prevent the following error when running the standalone binary:

```
pkg/prelude/bootstrap.js:1872
      throw error;
      ^

Error: Cannot find module '/snapshot/my_app/node_modules/axios/dist/node/axios.cjs'
1) If you want to compile the package/file into executable, please pay attention to compilation warnings and specify a literal in 'require' call. 2) If you don't want to compile the package/file into executable and want to 'require' it from filesystem (likely plugin), specify an absolute path in 'require' call using process.cwd() or process.execPath.
```

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/api.ts [hunk 1]
```diff
@@ -38,6 +39,10 @@ export class EndpointError extends Error {
 }
 
 export const formatBackendErrors = (requestError: AxiosError<BackendError>) => {
+  if (!requestError.config) {
+    return `config unavailable\n${requestError.message}`
+  }
```
Comment (jakepruitt) [inline src/commands/synthetics/api.ts:44]: Why this change? And why only for the syntehtics API client?

Comment (Drarig29) [reply src/commands/synthetics/api.ts:44]: It was to make TypeScript happy, as the `config` can now be `undefined`. But I agree, this adds a new possible error, so let's use `requestError.config?.baseURL` instead.

And only on Synthetics because it's the only place where we read properties from `AxiosError.config`

Comment (Drarig29) [reply src/commands/synthetics/api.ts]: Addressed in a1fb70da5b09aa69d7ac6b435a5421223cb338f7

**Review by juli1 — APPROVED** [review]

**Review by Letiste — APPROVED** [review]

**Review by romainkomorndatadog — APPROVED** [review]

File: jest.config-standalone.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-    'ts-jest': {
-      tsconfig: 'tsconfig.json',
-    },
+  transform: {
+    '^.+\\.tsx?$': [
+      'ts-jest',
+      {
+        tsconfig: 'tsconfig.json',
+      },
+    ],
```
Comment (etnbrd) [inline jest.config-standalone.js:9]: Why this change?

**Review by etnbrd — APPROVED** [review]
LGTM 👍 
Still curious about the standalone jest config.

File: jest.config-standalone.js (hunk 2 above)
Comment (Drarig29) [reply jest.config-standalone.js:9]: For a moment I thought it was related to the first issue I had with axios: 
image

Turns out it was _really_ just a warning, but it's still a good thing to fix 😄
